src/qc/calculate_atac_qc_metrics/script.py

```diff
@@ -0,0 +1,103 @@
+import sys
+
+import scanpy as sc
+import muon as mu
+from muon import atac as ac  # the module containing function for scATAC data processing
+import numpy as np
+
+## VIASH START
+par = {
+    "input": "resources_test/pbmc_1k_protein_v3/pbmc_1k_protein_v3_filtered_feature_bc_matrix.h5mu",
+    "fragments_path": None,
+    "output": "foo.h5mu",
+    "modality": "atac",
+    "layer": None,
+    "n_fragments_for_nucleosome_signal": 10e4,
+    "n_tss": 3000,
+    "nuc_signal_threshold": 2,
+    "tss_threshold": 1.5,
+}
+## VIASH END
+
+sys.path.append(meta["resources_dir"])
+# START TEMPORARY WORKAROUND setup_logger
+# reason: resources aren't available when using Nextflow fusion
+# from setup_logger import setup_logger
+def setup_logger():
+    import logging
+    from sys import stdout
+
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    console_handler = logging.StreamHandler(stdout)
+    logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
+    console_handler.setFormatter(logFormatter)
+    logger.addHandler(console_handler)
+
+    return logger
+# END TEMPORARY WORKAROUND setup_logger
+logger = setup_logger()
+
+def main():
+    logger.info("Reading input data")
+    mdata = mu.read(par["input"])
+
+    logger.debug("Making var names unique")
+    mdata.var_names_make_unique()
+    atac = mdata.mod[par["modality"]]
+
+    logger.info("Checking if QC columns are already calculated")
+    for col in ("n_features_per_cell", "total_fragment_counts"):
+        if col in atac.obs:
+            logger.warning(f"{col} is already in atac.obs, dropping")
+            atac.obs.drop(col, axis=1, inplace=True)
+    
+    logger.info("Calculating QC metrics")
+    sc.pp.calculate_qc_metrics(atac, percent_top=None, log1p=False, inplace=True, layer=par["layer"])
```

Comment:
I want to that things get overwritten by scanpy, could you pass a copy of the data?